03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,64 +1,61 @@
 const Task = require("../models/Task");
 
-const getAllTasks = async (req, res) => {
-  try {
-    const tasks = await Task.find({});
-    res.status(200).json({ tasks });
-  } catch (error) {
-    res.status(500).json({ msg: error });
-  }
+// asyncWrapper function for mistakes
+const asyncWrapper = (fn) => {
+  return async (req, res, next) => {
+    try {
+      await fn(req, res, next);
+    } catch (error) {
+      next(error);
+    }
+  };
 };
 
-const createTask = async (req, res) => {
-  try {
-    const task = await Task.create(req.body);
-    res.status(201).json({ task });
-  } catch (error) {
-    res.status(500).json({ msg: error });
-  }
-};
+const getAllTasks = asyncWrapper(async (req, res) => {
+  const tasks = await Task.find({});
+  res.status(200).json({ tasks });
+});
 
-const getTask = async (req, res) => {
-  try {
-    const { id: taskID } = req.params;
-    const task = await Task.findOne({ _id: taskID });
-    if (!task) {
-      return res.status(404).json({ msg: `No task with id: ${taskID}` });
-    }
-    res.status(200).json({ task });
-  } catch (error) {
-    res.status(500).json({ msg: error });
+const createTask = asyncWrapper(async (req, res) => {
+  const task = await Task.create(req.body);
+  res.status(201).json({ task });
+});
+
+const getTask = asyncWrapper(async (req, res, next) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOne({ _id: taskID });
+  if (!task) {
+    const error = new Error(`No task with id: ${taskID}`);
+    error.statusCode = 404;
+    return next(error);
   }
-};
+  res.status(200).json({ task });
+});
 
-const updateTask = async (req, res) => {
-  try {
-    const { id: taskID } = req.params;
-    const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
-      new: true,
-      runValidators: true,
-    });
-    if (!task) {
-      return res.status(404).json({ msg: `No task with id: ${taskID}` });
-    }
-    res.status(200).json({ task });
-  } catch (error) {
-    res.status(500).json({ msg: error });
+const updateTask = asyncWrapper(async (req, res, next) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
+    new: true,
+    runValidators: true,
+  });
+  if (!task) {
+    const error = new Error(`No task with id: ${taskID}`);
+    error.statusCode = 404;
```

Comment:
In the video, he creates a subclass that extends Error. I think he probably just wants you to be aware that you can extend existing classes and if you do you need to call super first in the constructor and that you can use `instanceof` to possibly demarcate between different class objects.